com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -8,101 +8,362 @@
+        /// For multi-instance unit tests, set this to true if you are use the <see cref="NetworkSceneManager"/>
         /// </summary>
-        public event NotifyClientAllClientsLoadedSceneDelegate OnNotifyClientAllClientsLoadedScene;
+        internal static bool IsTesting;
+
+        /// <summary>
+        /// The delegate callback definition for scene event notifications
+        /// For more details review over <see cref="SceneEvent"/> and <see cref="SceneEventData"/>
+        /// </summary>
+        /// <param name="sceneEvent"></param>
+        public delegate void SceneEventDelegate(SceneEvent sceneEvent);
+
+        /// <summary>
+        /// Event that will notify the local client or server of all scene events that take place
+        /// For more details review over <see cref="SceneEvent"/>, <see cref="SceneEventData"/>, and <see cref="SceneEventData.SceneEventTypes"/>
+        /// </summary>
+        public event SceneEventDelegate OnSceneEvent;
 
         internal readonly HashSet<string> RegisteredSceneNames = new HashSet<string>();
         internal readonly Dictionary<string, uint> SceneNameToIndex = new Dictionary<string, uint>();
         internal readonly Dictionary<uint, string> SceneIndexToString = new Dictionary<uint, string>();
-        internal readonly Dictionary<Guid, SceneSwitchProgress> SceneSwitchProgresses = new Dictionary<Guid, SceneSwitchProgress>();
-        internal readonly Dictionary<uint, NetworkObject> ScenePlacedObjects = new Dictionary<uint, NetworkObject>();
-
-        private static Scene s_LastScene;
-        private static string s_NextSceneName;
-        private static bool s_IsSwitching = false;
-        internal static uint CurrentSceneIndex = 0;
-        internal static Guid CurrentSceneSwitchProgressGuid = new Guid();
+        internal readonly Dictionary<Guid, SceneEventProgress> SceneEventProgressTracking = new Dictionary<Guid, SceneEventProgress>();
+
+        /// <summary>
+        /// Used to track in-scene placed NetworkObjects
+        /// We store them by:
+        /// [GlobalObjectIdHash][Scene.Handle][NetworkObject]
+        /// The Scene.Handle aspect allows us to distinguish duplicated in-scene placed NetworkObjects created by the loading
+        /// of the same additive scene multiple times.
+        /// </summary>
+        internal readonly Dictionary<uint, Dictionary<int, NetworkObject>> ScenePlacedObjects = new Dictionary<uint, Dictionary<int, NetworkObject>>();
+
+        /// <summary>
+        /// This is used for the deserialization of in-scene placed NetworkObjects in order to distinguish duplicated in-scene
+        /// placed NetworkObjects created by the loading of the same additive scene multiple times.
+        /// </summary>
+        internal Scene SceneBeingSynchronized;
+
+        // Used for observed object synchronization
+        private readonly List<NetworkObject> m_ObservedObjects = new List<NetworkObject>();
```

Comment:
I'm on a set kick today, and this could be another usage

Reply:
That is awesome!  By making that suggestion you made me realize that I "coded the need for this list away".  
So, my answer to this is that it no longer exists.  :)